### DISCUSSION TITLE
Junicode in Mediawiki

### DISCUSSION THREAD

**Discussion by redheadkelly** [psb1558/Junicode-font#256] [Q&A] [answered]
I'm attempting to use the Junicode font in my Mediawiki installation. I would like to use it only for text within articles that has been transcribed from old manuscripts. From what I can gather, I can access the Junicode font via Extension:UniversalLanguageSelector. 

If I load this extension, it does appear that the font will display using a CSS class with 'font-family: Junicode;'.

I would like to use the ss02 character set. I tried to use the 'lang="Irish"' setting.

For instance:
`<span class="junicode" lang="Irish"></span>`

It doesn't seem to do anything.

I can't figure out how to access specific characters, such as the insular t with dot above. I can see in the Irish_text.pdf included with the Junicode documentation, but I can't figure out how to get Mediawiki to display it.

Is this possible in Mediawiki?

Thanks.

**Comment by psb1558** [comment]
I don't know much about Mediawiki. Perhaps try the HTML language code for Irish: "ga"?

Dotted t is `U+1E6B`, or you can use the sequence `t U+0307`. When the language is set correctly and `ss02` is on, it will display as dotted insular t.

**Reply by redheadkelly** [reply]
Hi Peter. Thanks for replying.

Do you mind just helping me get it working in a regular HTML doc? Then maybe I can translate that into Mediawiki.

I'm working on a manuscript in old Irish, so I want to use ss02 in Irish, just like in the Irish_test.pdf.

Here's a [list of Junicode characters](https://rosssurnameproject.net/charts/junicode_characters.html) that, in general, won't render just by using the hex character. They clearly need additional settings. This list includes some of the ss02 characters. I added the small t with dot above, which displays fine on its own, to see if it would switch into the insular version with the right settings. I can't figure out what those setting should be.

Thanks. :)

**Comment by psb1558** [comment, marked as answer]
Okay, I understand your question now. You're asking about a number of recently added MUFI characters that are not yet in Junicode. Don't worry--they will appear. In the meantime, the vast majority of the characters in your list can already be produced by Junicode's preferred method, which avoids Private Use Area characters for reasons of accessibility and searchability. For example, U+F094 insular t with dot is not recognized by applications as a **t**. But if you include this rule in your CSS
```
            .insular {
                font-feature-settings: "ss02";
            }
```
and then include the attribute `class="insular"` in whatever element contains your Irish text (e.g. `<div class="insular" lang="ga">`), then the sequence `t&#x0307;` (t followed by combining dot) will produce an insular **t** with dot above:
<img width="40" alt="image" src="https://github.com/psb1558/Junicode-font/assets/16215767/6c191c8f-f391-49f0-98da-2774de014a75">
Unlike U+F094, this letter can be searched as a **t**, and it will behave like a normal letter in all respects (for example, it will kern properly with surrounding letters). Further, if your text lands in an environment where Junicode is not available (for example, if a reader copies the text and pastes it into a word processor), it will still be perfectly legible.

The [Junicode Manual](https://github.com/psb1558/Junicode-font/blob/master/docs/JunicodeManual.pdf) takes a very deep dive into these matters, and while it is far from a scintillating read, it contains lots of essential information for users.

Catching up with MUFI is very high on my to do list. But I am in the very odd position of including these Private Use Area characters in Junicode and at the same time strongly recommending that users avoid them.

**Reply by psb1558** [reply]
Junicode 2.204, released today, has fully caught up with MUFI.

**Reply by redheadkelly** [reply]
Thanks so much for this. I haven't been online this week so it was nice to return and find all of this. I did spend several days on this & go over the manual and several other sites before I asked, but was not able to piece this together for some reason, so I appreciate you taking the time to explain it.
